File: src/main/java/yunas/api/Batch.java
package yunas.api;

import kotlin.Unit;
import kotlin.jvm.functions.Function1;

/**
 * Batch Interface for Java.
 */
public interface Batch extends Function1<String[], Unit> {

    public void action(String[] args);

    @Override
    default Unit invoke(String[] args) {
        action(args);
        return null;
    }
}

File: src/main/kotlin/yunas/Batch.kt
package yunas

import java.util.function.Consumer

/**
 * Batch Interface for Java.
 */
@FunctionalInterface
interface Batch : Consumer<Array<String>> {

    fun action(args: Array<String>)

    override fun accept(args: Array<String>) {
        action(args)
    }
}
